tooling/templatize/pkg/pipeline/shell.go

```diff
@@ -82,16 +83,31 @@ func runShellStep(s *ShellStep, ctx context.Context, kubeconfigFile string, opti
 	logger.V(5).Info(fmt.Sprintf("Executing shell command: %s\n", s.Command), "command", s.Command)
 	output, err := cmd.CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("failed to execute shell command: %s %w", string(output), err)
+		return nil, fmt.Errorf("failed to execute shell command: %s %w", string(output), err)
 	}
 
-	s.outputFunc(string(output))
+	stringOutput := string(output)
+	return &stringOutput, nil
+}
 
+func runShellStepAndCaptureOutput(s *types.ShellStep, ctx context.Context, kubeconfigFile string, options *PipelineRunOptions, inputs map[string]output) error {
```

Comment:
I don't really understand why this function is necessary. `runShellStep` returns the output - why does a caller of `runShellStepAndCaptureOutput` want that output to be written to a file, just to be read back out of it? Why can't they just consume the output of the function in the first place?